cpp/src/writer/tsfile_writer.cc

```diff
@@ -419,6 +479,8 @@ int TsFileWriter::flush_chunk_group(MeasurementSchemaGroup *chunk_group) {
                        chunk_writer->get_chunk_data()))) {
         } else if (RET_FAIL(io_writer_->end_flush_chunk(
                        chunk_writer->get_chunk_statistic()))) {
+        } else {
+            chunk_writer->destroy();
         }
```

Comment:
If the return value indicates an error, it should be returned immediately.

Reply:
OK